src/CollapsibleField.js

```diff
@@ -241,35 +292,59 @@ class CollapsibleField extends Component {
 
   render() {
     let {
-      schema: { title },
+      schema: { title, properties = {} },
       uiSchema,
       registry: { fields },
       idSchema: { $id } = {},
       name,
-      formContext
+      formContext,
+      formData,
+      onChange: propsOnChange
     } = this.props;
     let { collapsed, AddElement } = this.state;
-    let { collapse: { field } } = uiSchema;
+    let { collapse: { collapsibleHeaderElements = {}, field } } = uiSchema;
     let CollapseElement = fields[field];
+    let { elements = [] } = collapsibleHeaderElements;
     // uischema retains the value form the state
     uiSchema.collapse.collapsed = this.state.collapsed;
 
     title = uiSchema["ui:title"] ? uiSchema["ui:title"] : title ? title : name;
     let customizedId = collapsed ? $id : undefined;
+
+    //remove header elements from the schema
+    let headerElementsSchemas = {};
+    let propertiesNoHeader = { ...properties };
```

Comment:
I remember talking about this. We could probably remove the spread here, right?

Reply:
Tried to remove this one but it caused a problem.  Apparently deleting the key from the actual schema property object causes the checkbox to disappear on rerender.  By making a shallow copy of the properties object the original still has a reference to the checkbox schema for rerender